go/pkg/services/logcollector/server.go

```diff
@@ -321,17 +339,27 @@ func (s *LogCollectorServer) startLogStreaming(ctx context.Context,
 		streamErr error
 		stream    io.ReadCloser
 	)
+	defer stream.Close() // nolint: errcheck
 
 	// stream logs - retry if failed
-	for {
+	err := common.RetryUntilSuccessful(5*time.Second, 1*time.Second, func() (bool, error) {
 		stream, streamErr = restClientRequest.Stream(ctx)
-		if streamErr == nil {
-			break
+		if streamErr != nil {
+			s.Logger.WarnWithCtx(ctx,
+				"Failed to get pod log stream",
+				"runUID", runUID,
+				"err", streamErr)
+			return true, streamErr
 		}
-		s.Logger.WarnWithCtx(ctx, "Failed to get pod log stream read/closer, retrying", "runUID", runUID, "err", streamErr.Error())
-		time.Sleep(1 * time.Second)
+		return false, nil
+	})
+	if err != nil {
+		s.Logger.ErrorWithCtx(ctx,
+			"Failed to get pod log stream",
+			"runUID", runUID,
+			"err", err)
+		return
```

Comment:
Didn't we want this retry to go on indefinitely on purpose?

Reply:
Hmmm but why tho? What if it never succeeds?
The way this works now is that after this goroutine exists, the log collection monitor will call `StartLog` again for this runUID, so the whole procedure will restart.